Formula/vlang.rb

```diff
@@ -0,0 +1,19 @@
+class Vlang < Formula
+  desc "V programming language"
+  homepage "https://vlang.io"
+  url "https://github.com/vlang/v/archive/0.1.20.tar.gz"
+  sha256 "8102b48b2c82be6be14633e76e71e215aab5221198315436f97be53e1abe1f5d"
+
+  def install
+    system "make"
+    bin.install "v"
+    bin.install "vlib","compiler","examples","thirdparty","tools"
+  end
+
+  test do
+    version_output = shell_output("#{bin}/v -v")
+    assert_match version.to_s, version_output
+
+    assert_match "OK", shell_output("#{bin}/v test v")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
`v test v` performs functionality tests, like make check. It is stronger than testing simple program.

Reply:
We're not trying to check if the program was written correctly though. We're trying to check if a user can use the software. I'd much rather see an actual use case being tested.

Reply:
I'm not sure if this idea is the best one but my suggestion would be adding a new command to the `V` CLI that executes `V` code. 

For example:
``` 
Options/commands:
  -h, help          Display this information.
  -e 'command'      one line of script. Several -e's allowed.
```

then we'd be able to test the behaviour here:
```
any_v_command = shell_output("#{bin}/v -e 'print("test")'")
assert_equal "test", any_v_command
```

Thoughts?

Reply:
V is a compiler. Its job is to compile source code. `v test v` does just that. In fact it compiles and executes over 52 test programs, and compiles (without executing) over 19 example programs.

Reply:
After running `v test v` there are artifacts (binaries) created inside examples/ , for example examples/news_fetcher is produced by compiling examples/news_fetcher.v , examples/hello_world is produced by compiling examples/hello_world.v .

An individual file compilation can be done simply by doing:
`v examples/hello_world.v` then checking to see whether examples/hello_world is produced, is executable and can run .

Reply:
although full functionality test is waiting for https://github.com/vlang/v/issues/2176, basic test is already available.